webapp/store/views.py

```diff
@@ -57,59 +60,93 @@ def search_redirect(path=None):
+        query_string.append(str(request.query_string, "utf-8"))
+    return redirect("/search?{}".format("&".join(query_string)), code=302)
 
 
-@jaasstore.route('/u/<username>/')
+@jaasstore.route("/u/<username>/")
 def user_details(username):
     entities = models.get_user_entities(username)
-    if len(entities['charms']) > 0 or len(entities['bundles']) > 0:
+    if len(entities["charms"]) > 0 or len(entities["bundles"]) > 0:
         return render_template(
-            'store/user-details.html',
+            "store/user-details.html",
             context={
-                'bundles_count': len(entities['bundles']),
-                'bundles': entities['bundles'],
-                'charms_count': len(entities['charms']),
-                'charms': entities['charms'],
-                'entities': entities,
-                'username': username
-            }
+                "bundles_count": len(entities["bundles"]),
+                "bundles": entities["bundles"],
+                "charms_count": len(entities["charms"]),
+                "charms": entities["charms"],
+                "entities": entities,
+                "username": username,
+            },
         )
     else:
         return abort(404, "User not found: {}".format(username))
 
 
-@jaasstore.route('/u/<username>/<charm_or_bundle_name>')
-@jaasstore.route(
-    '/u/<username>/<charm_or_bundle_name>/<series_or_version>')
-@jaasstore.route(
-    '/u/<username>/<charm_or_bundle_name>/<series_or_version>/<version>')
-def user_entity(username, charm_or_bundle_name, series_or_version=None,
-                version=None):
-    return details(charm_or_bundle_name, series_or_version=series_or_version,
-                   version=version)
+@jaasstore.route("/u/<username>/<entity_name>/")
+def user_entity(username, entity_name):
+    reference = references.Reference.from_jujucharms_url(request.path[1:])
+    entity = models.get_charm_or_bundle(reference)
+
+    if entity:
+        if entity["is_charm"]:
+            # entity["display_name"] = ""
+            # entity["revision_number"] = ""
+            # entity["user"] = ""
+            # entity["latest_revision_id"] = ""
+            # entity["series"] = ""
+            # entity["channel_details"] = ""
+            entity["description"] = entity["charm_data"]["Meta"]["charm-metadata"][
+                "Description"
+            ]
+            entity["user"] = entity["charm_data"]["Meta"]["owner"]["User"]
+            return render_template(
+                "store/charm-details.html", context={"entity": entity}
+            )
+        else:
+            entity["user"] = entity["bundle_data"]["Meta"]["owner"]["User"]
+            entity["id"] = entity["bundle_data"]["Id"]
+            entity["meta_published_info"] = entity["bundle_data"]["Meta"]["published"][
+                "Info"
+            ]
+            return render_template(
+                "store/bundle-details.html", context={"entity": entity}
```

Comment:
It's not at all clear here what data is available to the template. The only way to really find out is to run the query and log it out.

Something like https://github.com/juju/jaaslibjs/blob/master/lib/charmstore.js#L170 would clear this up here. So that every piece of data is explicitly called out.

Reply:
I'm not sure what you mean - can you make the change you suggest please and push it to this branch?